src/nvim/keymap.c

```diff
@@ -665,19 +662,12 @@ int extract_modifiers(int key, int *modp)
     modifiers &= ~MOD_MASK_SHIFT;
   }
   if ((modifiers & MOD_MASK_CTRL)
-      && ((key >= '?' && key <= '_') || ASCII_ISALPHA(key))
-      ) {
+      && ((key >= '?' && key <= '_') || ASCII_ISALPHA(key))) {
     key = Ctrl_chr(key);
     modifiers &= ~MOD_MASK_CTRL;
-    /* <C-@> is <Nul> */
-    if (key == 0)
+    if (key == 0) {  // <C-@> is <Nul>
       key = K_ZERO;
-  }
-  if ((modifiers & MOD_MASK_ALT) && key < 0x80
-      && !enc_dbcs                      /* avoid creating a lead byte */
-      ) {
-    key |= 0x80;
-    modifiers &= ~MOD_MASK_ALT;         /* remove the META modifier */
```

Comment:
By deleting this block, 

- we no longer set the high bit to represent alt
- the global `mod_mask` keeps the `MOD_MASK_ALT` flag